CardPayments/src/test/java/com/paypal/android/cardpayments/CardAuthLauncherUnitTest.kt

```diff
@@ -0,0 +1,274 @@
+package com.paypal.android.cardpayments
+
+import android.net.Uri
+import androidx.fragment.app.FragmentActivity
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchException
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.BrowserSwitchStatus
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CardAuthLauncherUnitTest {
+
+    private lateinit var browserSwitchClient: BrowserSwitchClient
+    private lateinit var sut: CardAuthLauncher
+
+    // TODO: consider using androidx.test activity instead of mockk
+    // Ref: https://robolectric.org/androidx_test/#activities
+    private val activity: FragmentActivity = mockk(relaxed = true)
+
+    private val url = Uri.parse("https://fake.com/destination")
+    private val card = Card("4111111111111111", "01", "25", "123")
+
+    private val approveOrderMetadata = JSONObject()
+        .put("request_type", "approve_order")
+        .put("order_id", "fake-order-id")
+
+    private val vaultMetadata = JSONObject()
+        .put("request_type", "vault")
+        .put("setup_token_id", "fake-setup-token-id")
+
+    @Before
+    fun beforeEach() {
+        browserSwitchClient = mockk(relaxed = true)
+    }
+
+    @Test
+    fun `presentAuthChallenge() returns an error for approve order when it cannot browser switch`() {
+        val browserSwitchException = mockk<BrowserSwitchException>()
+        every { browserSwitchException.message } returns "error message from browser switch"
+
+        every { browserSwitchClient.start(any(), any()) } throws browserSwitchException
+
+        val returnUrl = "merchant.app://return.com/deep-link"
+        val cardRequest = CardRequest("fake-order-id", card, returnUrl)
+        val cardAuthChallenge = CardAuthChallenge.ApproveOrder(url, cardRequest)
+
+        sut = CardAuthLauncher(browserSwitchClient)
+        val error = sut.presentAuthChallenge(activity, authChallenge = cardAuthChallenge)
+        assertEquals("error message from browser switch", error?.errorDescription)
```

Comment:
^ These tests were extracted from `CardClientUnitTest`. Eventually the launcher will be separated from the client in the next major version.